components/forwarder/envelope.go

```diff
@@ -0,0 +1,74 @@
+package forwarder
+
+import (
+	"encoding/json"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/pkg/errors"
+)
+
+// messageEnvelope wraps Watermill message and contains destination topic.
+type messageEnvelope struct {
+	DestinationTopic string `json:"destination_topic"`
+
+	UUID     string            `json:"uuid"`
+	Payload  []byte            `json:"payload"`
+	Metadata map[string]string `json:"metadata"`
+}
+
+func newMessageEnvelope(destTopic string, uuid string, payload []byte, metadata map[string]string) (*messageEnvelope, error) {
+	e := &messageEnvelope{
+		DestinationTopic: destTopic,
+		UUID:             uuid,
+		Payload:          payload,
+		Metadata:         metadata,
+	}
+
+	if err := e.validate(); err != nil {
+		return nil, errors.Wrap(err, "cannot create a message envelope")
+	}
+
+	return e, nil
+}
+
+func (e *messageEnvelope) validate() error {
+	if e.DestinationTopic == "" {
+		return errors.New("unknown destination topic")
+	}
+	if e.UUID == "" {
```

Comment:
well, in theory, you can have watermill message with empty UUID ;)

Reply:
Is it correct scenario then? I don't know what would that mean.

Reply:
yes, this is correct scenario

Reply:
00a0ad8